tests/test_base.py

```diff
@@ -636,3 +638,256 @@ def test_pack_header_for_user(self, app, user_class):
             assert token_data['rls'] == 'admin,operator'
             assert token_data['duder'] == 'brief'
             assert token_data['el_duderino'] == 'not brief'
+
+    def test_registration_email(
+        self, app, user_class, db, clean_flask_app_config
+    ):
+        """
+        This test verifies email based registration functions as expected.
+        This includes sending messages with valid time expiring JWT tokens
+           and ensuring the body matches the expected body, as well
+           as token validation.
+        """
+
+        _pwd = dirname(dirname(abspath(__file__)))
+        here = _pwd + '/flask_praetorian/templates'
+        tmpl_file = here + '/registration_email.html'
+        with open(tmpl_file) as _template:
+            tmpl = Template(_template.read())
+
+        # generate (w/o sending) a notification email and match it to our own
+        app.config['TESTING'] = True
+        app.config['PRAETORIAN_EMAIL_TEMPLATE'] = tmpl_file
+        app.config['PRAETORIAN_CONFIRMATION_ENDPOINT'] = 'unprotected'
+
+        default_guard = Praetorian(app, user_class)
+
+        # create our default test user
+        the_dude = user_class(
```

Comment:
we should make this a fixture that handles setup and cleanup on its own

Reply:
Can leave this for another pull?

Reply:
Sure. Please create an issue.